### PR TITLE
Ongoing AML

### PR DESCRIPTION
**Pull request by alonp99**
### Description
Elaborate on the subject, motivation, and context.

### Related issues
 * Provide a link to each related issue.

### Breaking changes
 * Describe the breaking changes that this pull request introduces.

### How these changes were tested
 * Describe the tests that you ran to verify your changes, including devices, operating systems, browsers and versions.

### Examples and references
 * Links, screenshots, and other resources related to this change.

### Checklist
- [] I have read the [contribution guidelines](CONTRIBUTING.md) of this project
- [] I have read the [style guidelines](STYLE_GUIDE.md) of this project
- [] I have performed a self-review of my own code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings and errors
- [] New and existing tests pass locally with my changes

___

## **Type**
enhancement, bug_fix

___

## **Description**
- Added new actions (`DISMISS`, `FLAG`) for case management.
- Implemented mutation hooks (`useDismissCaseMutation`, `useFlagCaseMutation`) for dismissing and flagging cases.
- Refactored existing hooks and components to accommodate new case management actions.
- Added `OnGoingBlocks` component and logic for handling ongoing cases.
- Implemented `ActionsVariant` component to handle different action variants based on the workflow definition.
- Added success and error toast messages for new actions.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><details><summary>24 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>enums.ts</strong><dd><code>Add New Actions for Case Management</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/backoffice-v2/src/common/enums.ts
- Added `DISMISS` and `FLAG` actions to the `Action` enum.

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2144/files#diff-a399fea7c08413cee510703f5a7eefe68a11109588ed3c1d646598cbeedccc30">+2/-0</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>useApproveCaseMutation.tsx</strong><dd><code>Refactor useApproveCaseMutation Hook Parameter Name</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/backoffice-v2/src/domains/entities/hooks/mutations/useApproveCaseMutation/useApproveCaseMutation.tsx
<li>Changed parameter name from <code>onSelectNextEntity</code> to <code>onSelectNextCase</code>.<br>

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2144/files#diff-d9f0502a51ff877d157ed2f05cba21632adeb68536f1ff9d714dcf7c703cebf1">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>useDismissCaseMutation.tsx</strong><dd><code>Implement Dismiss Case Mutation Hook</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/backoffice-v2/src/domains/entities/hooks/mutations/useDismissCaseMutation/useDismissCaseMutation.tsx
- Created `useDismissCaseMutation` hook for dismissing cases.

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2144/files#diff-77f62e9b095eb89c9762a2c560f511a4ca62a1ebbfb62daa67bc6bd29c391b80">+38/-0</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>useFlagCaseMutation.tsx</strong><dd><code>Implement Flag Case Mutation Hook</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/backoffice-v2/src/domains/entities/hooks/mutations/useFlagCaseMutation/useFlagCaseMutation.tsx
- Created `useFlagCaseMutation` hook for flagging cases.

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2144/files#diff-efe707722a1b252de3bc5aebb9af45b5198c80ca3c95239d1563255e226f67cb">+38/-0</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>useRejectCaseMutation.tsx</strong><dd><code>Rename Reject Case Mutation Hook</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/backoffice-v2/src/domains/entities/hooks/mutations/useRejectCaseMutation/useRejectCaseMutation.tsx
- Renamed `useRejectEntityMutation` to `useRejectCaseMutation`.

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2144/files#diff-2c1f551bd41b6fd8fc519950080ced3444b3869b4675c7c54302671ccdfee754">+3/-3</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>useSelectNextCase.tsx</strong><dd><code>Rename Select Next Case Hook</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/backoffice-v2/src/domains/entities/hooks/useSelectNextCase/useSelectNextCase.tsx
- Renamed `useSelectNextEntity` to `useSelectNextCase`.

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2144/files#diff-e57655ad60368f06dc6ee02df478ee619c4a5f3c683b79bd16461daacd76b176">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>fetchers.ts</strong><dd><code>Add Mock Data for KYC Session AML Hits</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/backoffice-v2/src/domains/workflows/fetchers.ts
- Added mock data for KYC session with AML hits.

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2144/files#diff-2e11dd9637644881292fb6f179c35a4c66109f38d0019605b0789eed77042167">+47/-0</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>useAmlBlock.tsx</strong><dd><code>Implement AML Block Rendering Hook</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/backoffice-v2/src/lib/blocks/components/AmlBlock/hooks/useAmlBlock/useAmlBlock.tsx
- Implemented `useAmlBlock` hook for AML block rendering.

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2144/files#diff-1c50004d8f75354da75513b2439ab2321b17b17fc38cf457ed1d42f6162317c0">+377/-0</a>&nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>aml-adapter.ts</strong><dd><code>Add AML Data Adapter Utility Function</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/backoffice-v2/src/lib/blocks/components/AmlBlock/utils/aml-adapter.ts
- Added `amlAdapter` utility function for AML data adaptation.

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2144/files#diff-87cf550e154951707d6c89cef48422e8fc4323e0d4eb0798058220998bf70e56">+81/-0</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>useKycBlock.tsx</strong><dd><code>Refactor KYC Block to Use AML Block Hook</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/backoffice-v2/src/lib/blocks/components/KycBlock/hooks/useKycBlock/useKycBlock.tsx
- Refactored to use `useAmlBlock` hook.

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2144/files#diff-228217e3471e5da9a0eb2a1c7a54fd6e61adc2789fd66e3ca7de6824a5bfb0aa">+12/-432</a></td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>NoBlocks.tsx</strong><dd><code>Adjust Padding for NoBlocks Component</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/backoffice-v2/src/lib/blocks/components/NoBlocks/NoBlocks.tsx
- Adjusted padding for the `NoBlocks` component.

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2144/files#diff-8fc49b9c4450f84bbb08496faf858e5067bc11701ed7c11d956df100d5bd7886">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>BlocksVariant.tsx</strong><dd><code>Add Handling for Ongoing Variant in BlocksVariant</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/backoffice-v2/src/lib/blocks/variants/BlocksVariant/BlocksVariant.tsx
- Added handling for the `ONGOING` variant.

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2144/files#diff-9dfa418edb7e78e25a01f856606ee3cc4f462578c196dc18abc12588e853c85d">+7/-0</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>OnGoingBlocks.tsx</strong><dd><code>Implement OnGoingBlocks Component</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/backoffice-v2/src/lib/blocks/variants/OnGoingBlocks/OnGoingBlocks.tsx
- Implemented `OnGoingBlocks` component for ongoing cases.

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2144/files#diff-31ff7a253ed826d670f7d07ba21969c78b2acfdaa296e2e4d5f9df57f194c9c0">+17/-0</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>useOnGoingBlocksLogic.tsx</strong><dd><code>Implement Logic Hook for OnGoingBlocks Component</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/backoffice-v2/src/lib/blocks/variants/OnGoingBlocks/hooks/useOnGoingBlocksLogic/useOnGoingBlocksLogic.tsx
- Implemented logic hook for `OnGoingBlocks` component.

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2144/files#diff-922a42ccd7fce5e4e55222d1c979a11c8a88c2035fbd87c97195e32f91743233">+47/-0</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>variant-checkers.ts</strong><dd><code>Add Checker Function for Ongoing Variant</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/backoffice-v2/src/lib/blocks/variants/variant-checkers.ts
- Added checker function for `ONGOING` variant.

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2144/files#diff-1fc2a84611ee27084e653151776ca5719a98c3e84a92b6db120056af62dd3729">+6/-0</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>Case.Actions.tsx</strong><dd><code>Refactor Case Actions to Use ActionsVariant Component</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/backoffice-v2/src/pages/Entity/components/Case/Case.Actions.tsx
- Refactored to use `ActionsVariant` component.

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2144/files#diff-f3e71bf810a855cd82f26bfb67e942d73aa95a6001bfaa7c5a5264a2d9daa3bd">+10/-88</a>&nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ActionsVariant.tsx</strong><dd><code>Implement ActionsVariant Component</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/backoffice-v2/src/pages/Entity/components/Case/actions-variants/ActionsVariant/ActionsVariant.tsx
<li>Implemented <code>ActionsVariant</code> component for handling different action <br>variants.<br>

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2144/files#diff-3a6ddf7bed9d4b5d989d860708769a08fd51c1f7f28c9e66531777bcd40a687c">+17/-0</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>DefaultActions.tsx</strong><dd><code>Implement DefaultActions Component</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/backoffice-v2/src/pages/Entity/components/Case/actions-variants/DefaultActions/DefaultActions.tsx
- Implemented `DefaultActions` component for default case actions.

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2144/files#diff-f7c5903ddc2f807c43d8c50818d6bea426ee94404a086d8821fd755c2520e3c8">+98/-0</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>useDefaultActionsLogic.tsx</strong><dd><code>Implement Logic Hook for DefaultActions Component</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/backoffice-v2/src/pages/Entity/components/Case/actions-variants/DefaultActions/hooks/useDefaultActionsLogic/useDefaultActionsLogic.tsx
- Implemented logic hook for `DefaultActions` component.

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2144/files#diff-61e7582b5b95a047054c177022cb0cfb6a99ad1c772ae8c51e58d09a3f95bde0">+82/-0</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>OnGoingActions.tsx</strong><dd><code>Implement OnGoingActions Component</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/backoffice-v2/src/pages/Entity/components/Case/actions-variants/OnGoingActions/OnGoingActions.tsx
- Implemented `OnGoingActions` component for ongoing case actions.

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2144/files#diff-77c9b88ce830c4a3cbcc76b9ccf5613c83f828fe76939f3fee5668a9a5db371e">+110/-0</a>&nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>useOnGoingActionsLogic.tsx</strong><dd><code>Implement Logic Hook for OnGoingActions Component</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/backoffice-v2/src/pages/Entity/components/Case/actions-variants/OnGoingActions/hooks/useOnGoingActionsLogic/useOnGoingActionsLogic.tsx
- Implemented logic hook for `OnGoingActions` component.

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2144/files#diff-29cec2faaae4cd9532b556f76bb709310fd2f4a7a912f5a509c3c37925018782">+64/-0</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>useCaseActionsLogic.tsx</strong><dd><code>Refactor Case Actions Logic and Add Workflow Definition</code>&nbsp; &nbsp; </dd></summary>
<hr>

apps/backoffice-v2/src/pages/Entity/components/Case/hooks/useCaseActionsLogic/useCaseActionsLogic.tsx
<li>Refactored to remove unused imports and logic.<br> <li> Added <code>workflowDefinition</code> to the return object.<br>

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2144/files#diff-a0d1bfafe71d59a4be3cfc1eb89891bf2eac361942fae6e463b0ca0b296fa4e0">+1/-45</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>index.ts</strong><dd><code>Add Ongoing Variant to WorkflowDefinitionVariant</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/common/src/consts/index.ts
- Added `ONGOING` to `WorkflowDefinitionVariant`.

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2144/files#diff-5179492f8e46f2d336de4be5862de29dbd87a786238e443a9e21d96ba46ae1f1">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>toast.json</strong><dd><code>Add Toast Messages for Dismiss and Flag Actions</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/backoffice-v2/public/locales/en/toast.json
<li>Added success and error messages for dismissing and flagging cases.<br>

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2144/files#diff-d30ed0a56705e159998e269db1a077b7180515e77a34b6b4ad38948c3867f2df">+8/-0</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    
</table></details></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: packages/common/.eslintrc.cjs [hunk 1]
```diff
@@ -1,4 +1,7 @@
 /** @type {import('eslint').Linter.Config} */
 module.exports = {
   extends: ['@ballerine/eslint-config'],
+  parserOptions: {
+    project: './tsconfig.eslint.json',
+  },
```
Comment (Omri-Levy) [inline packages/common/.eslintrc.cjs:6]: Boyscout. Solves an error.

File: apps/backoffice-v2/src/domains/entities/hooks/mutations/useDismissCaseMutation/useDismissCaseMutation.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        },
+      }),
+    onSuccess: () => {
+      // workflowsQueryKeys._def is the base key for all workflows queries
+      void queryClient.invalidateQueries(workflowsQueryKeys._def);
+
+      toast.success(t('toast:dismiss_case.success'));
+
+      // TODO: Re-implement
+      // onSelectNextEntity();
```
Comment (tomer-shvadron) [inline apps/backoffice-v2/src/domains/entities/hooks/mutations/useDismissCaseMutation/useDismissCaseMutation.tsx:32]: Can you explain this?

File: apps/backoffice-v2/src/lib/blocks/components/AmlBlock/hooks/useAmlBlock/useAmlBlock.tsx [hunk 3]
```diff
@@ -0,0 +1,377 @@
...
+}: {
+  sessionKeys: string[];
+  getAmlData: (key: string) => Parameters<typeof amlAdapter>[0];
+}) => {
+  const isAmlEmpty = useMemo(() => {
+    if (!sessionKeys?.length) {
+      return true;
+    }
+
+    return safeEvery(sessionKeys, key => {
```
Comment (tomer-shvadron) [inline apps/backoffice-v2/src/lib/blocks/components/AmlBlock/hooks/useAmlBlock/useAmlBlock.tsx]:
```suggestion
    return safeEvery(sessionKeys, key => !getAmlData(key));
```

File: apps/backoffice-v2/src/lib/blocks/components/AmlBlock/hooks/useAmlBlock/useAmlBlock.tsx [hunk 4]
```diff
@@ -0,0 +1,377 @@
...
+      const aml = getAmlData(key);
+
+      if (!Object.keys(aml ?? {}).length) return [];
+
+      const { totalMatches, fullReport, dateOfCheck, matches } = amlAdapter(aml);
+
+      return [
+        ...createBlocksTyped()
+          .addBlock()
+          .addCell({
```
Comment (tomer-shvadron) [inline apps/backoffice-v2/src/lib/blocks/components/AmlBlock/hooks/useAmlBlock/useAmlBlock.tsx:23]: I think we should extract the cells into variables that can be placed in their own files. Right now this file reminds me of the old `useTasks()`, we can do better.

File: apps/backoffice-v2/src/lib/blocks/variants/BlocksVariant/BlocksVariant.tsx [hunk 5]
```diff
@@ -22,5 +25,9 @@ export const BlocksVariant: FunctionComponent<{
     return <ManualReviewBlocks />;
   }
 
+  if (isOnGoingVariant) {
```
Comment (tomer-shvadron) [inline apps/backoffice-v2/src/lib/blocks/variants/BlocksVariant/BlocksVariant.tsx]: I understand why you'd think I'm a grammer nazi, but Ongoing is a single word (not "On Going" or "On-Going"), so I'd expect `isOngoingVariant`, `<OngoingBlocks />` and so on.

File: apps/backoffice-v2/src/lib/blocks/variants/OnGoingBlocks/OnGoingBlocks.tsx [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import { BlocksComponent } from '@ballerine/blocks';
+import { NoBlocks } from '@/lib/blocks/components/NoBlocks/NoBlocks';
+import { cells } from '@/lib/blocks/create-blocks-typed/create-blocks-typed';
+import { useOnGoingBlocksLogic } from '@/lib/blocks/variants/OnGoingBlocks/hooks/useOnGoingBlocksLogic/useOnGoingBlocksLogic';
+
+export const OnGoingBlocks = () => {
+  const { blocks, isLoading } = useOnGoingBlocksLogic();
```
Comment (tomer-shvadron) [inline apps/backoffice-v2/src/lib/blocks/variants/OnGoingBlocks/OnGoingBlocks.tsx]:
```suggestion
  const { blocks, isLoading } = useOngoingBlocksLogic();
```

File: services/workflows-service/src/webhooks/webhooks.controller.ts [hunk 7]
```diff
@@ -0,0 +1,55 @@
+import * as common from '@nestjs/common';
+import * as swagger from '@nestjs/swagger';
+import * as nestAccessControl from 'nest-access-control';
+import * as errors from '../errors';
+import { Public } from '@/common/decorators/public.decorator';
+import { AmlWebhookInput } from './dtos/aml-webhook-input';
+import { IndividualAmlWebhookInput } from '@/webhooks/dtos/individual-aml-webhook-input';
+import { WebhooksService } from '@/webhooks/webhooks.service';
+
+const WEBHOOKS = {
```
Comment (Omri-Levy) [inline services/workflows-service/src/webhooks/webhooks.controller.ts]: Adhere to existing enum conventions.
```suggestion
const Webhook = {
```

File: apps/backoffice-v2/src/lib/blocks/variants/variant-checkers.ts [hunk 8]
```diff
@@ -14,3 +14,9 @@ export const checkIsManualReviewVariant = (
   workflowDefinition?.version >= 0 &&
   workflowDefinition?.variant === WorkflowDefinitionVariant.MANUAL_REVIEW &&
   workflowDefinition?.config?.isLegacyReject;
+
+export const checkIsOnGoingVariant = (
+  workflowDefinition: Pick<TWorkflowById['workflowDefinition'], 'variant' | 'config' | 'version'>,
+) =>
+  workflowDefinition?.version >= 0 &&
```
Comment (tomer-shvadron) [inline apps/backoffice-v2/src/lib/blocks/variants/variant-checkers.ts:21]: Why is the `version` check needed?

File: services/workflows-service/src/webhooks/webhooks.controller.ts [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+  AML_INDIVIDUAL_MONITORING_UPDATE: 'aml.individuals.monitoring.update',
+} as const;
+
+const ENTITY_TYPES = {
```
Comment (Omri-Levy) [inline services/workflows-service/src/webhooks/webhooks.controller.ts]: Adhere to existing enum conventions.
```suggestion
const EntityType = {
```

File: apps/backoffice-v2/src/pages/Entity/components/Case/actions-variants/OnGoingActions/OnGoingActions.tsx [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+import { DialogHeader } from '@/common/components/organisms/Dialog/Dialog.Header';
+import { DialogTitle } from '@/common/components/organisms/Dialog/Dialog.Title';
+import { DialogDescription } from '@/common/components/organisms/Dialog/Dialog.Description';
+import { DialogFooter } from '@/common/components/organisms/Dialog/Dialog.Footer';
+import { DialogClose } from '@radix-ui/react-dialog';
+import React from 'react';
+
+import { useOnGoingActionsLogic } from '@/pages/Entity/components/Case/actions-variants/OnGoingActions/hooks/useOnGoingActionsLogic/useOnGoingActionsLogic';
+
+export const OnGoingActions = () => {
```
Comment (tomer-shvadron) [inline apps/backoffice-v2/src/pages/Entity/components/Case/actions-variants/OnGoingActions/OnGoingActions.tsx]:
```suggestion
export const OngoingActions = () => {
```

**Review by tomer-shvadron — APPROVED** [review]

File: apps/backoffice-v2/src/lib/blocks/components/AmlBlock/hooks/useAmlBlock/useAmlBlock.tsx (hunk 4 above)
Comment (Omri-Levy) [reply apps/backoffice-v2/src/lib/blocks/components/AmlBlock/hooks/useAmlBlock/useAmlBlock.tsx:23]: Many cells would have to be functions. It is important to me not to mix variables and functions. I am also very interested in introducing a first-class pattern for cells/blocks re-use.

File: apps/backoffice-v2/src/lib/blocks/variants/variant-checkers.ts (hunk 8 above)
Comment (Omri-Levy) [reply apps/backoffice-v2/src/lib/blocks/variants/variant-checkers.ts:21]: Maybe doing so was a mistake but it was done as a showcase of the new direction and a placeholder/future change. Maybe redundant but there was a thought that each check should have the same baseline of these conditions.

File: apps/backoffice-v2/src/lib/blocks/components/AmlBlock/hooks/useAmlBlock/useAmlBlock.tsx (hunk 4 above)
Comment (tomer-shvadron) [reply apps/backoffice-v2/src/lib/blocks/components/AmlBlock/hooks/useAmlBlock/useAmlBlock.tsx:23]: What do you suggest to do so the file will be more readable and re-useable?

File: apps/backoffice-v2/src/domains/entities/hooks/mutations/useDismissCaseMutation/useDismissCaseMutation.tsx (hunk 2 above)
Comment (Omri-Levy) [reply apps/backoffice-v2/src/domains/entities/hooks/mutations/useDismissCaseMutation/useDismissCaseMutation.tsx:32]: Worth asking Nitzan about this again.

File: apps/backoffice-v2/src/lib/blocks/components/AmlBlock/hooks/useAmlBlock/useAmlBlock.tsx (hunk 4 above)
Comment (Omri-Levy) [reply apps/backoffice-v2/src/lib/blocks/components/AmlBlock/hooks/useAmlBlock/useAmlBlock.tsx:23]: Will have to give this more thought.

**Pull request merged**